app.go

```diff
@@ -24,142 +24,238 @@ import (
 	"context"
 	"os"
 	"os/signal"
-	"reflect"
 	"syscall"
 	"time"
 
-	"github.com/pkg/errors"
 	"go.uber.org/dig"
 	"go.uber.org/fx/internal/fxlog"
 	"go.uber.org/fx/internal/fxreflect"
 	"go.uber.org/multierr"
 )
 
-// App models a modular application
+// DefaultTimeout is the start and stop timeout used by Run.
+const DefaultTimeout = 15 * time.Second
+
+// Timeout is a convenience function to construct a context with a timeout.
+// It's only intended to reduce noise in the main function; since it doesn't
+// expose context.CancelFunc, it may leak resources.
+func Timeout(d time.Duration) context.Context {
```

Comment:
Good point. The team's general perception seems to be that the native context API is wordy, so the difference between using `Run` and `Start`/`Stop` is large. This was an attempt to reduce that delta, at least in `main`, without introducing functional options for `Start` and `Stop`.